src/ui/tabs/NestedTab.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/overlay.module.css';
+
+function NestedTab(props) {
+  const {
+    active, children, onClick, parentTabs, title,
+  } = props;
+
+  // TODO: be smarter about using the active property to reurn content
+  // Maybe we could separate into active (aka display this content) vs isHeader?
+  if (!active) {
+    throw new Error('At least one NestedTab must be active');
+  }
+
+  if (!children) {
+    return null;
+  }
+
+  const nestedTabs = (
+    <span className={styles.NestedTabHeader}>
+      {parentTabs}
+      {parentTabs ? <span>{'>'}</span> : null}
+      <button
+        className={styles.NestedTabTitle}
+        type="button"
+        tabIndex={0}
+        onClick={onClick}
+        onKeyPress={onClick}
+      >
+        {title}
+      </button>
+    </span>
+  );
+
+  const nestedContent = () => {
+    const content = [nestedTabs];
+    const childrenArray = React.Children.toArray(props.children);
+    for (let i = 0; i < childrenArray.length; i += 1) {
```

Comment:
2 points:

1. Conventional for loops (`for (<init>; <condition>; <increment>)`) are actually rarely used in JS, especially when iterating over collections. There are 2 ways that you will typically see iteration done:

    `for-of` loop:
    ```js
    for (const child of React.Children.toArray(children)) {
        ...
    }
    ```

    `Array.forEach`:
    ```js
    React.Children.toArray(children).forEach(child => {
        ...
    });
    ```

2. `React.Children` actually has a shorthand for this:
    ```js
    React.Children.forEach(children, child => {
        ...
    });
    ```

Reply:
Although I see now that you have a `return` inside the loop, so the `for-of` loop is the only viable option here.